material/plugins/privacy/plugin.py

```diff
@@ -404,18 +410,28 @@ def _fetch(self, file: File, config: MkDocsConfig):
 
             # Download external asset
             log.info(f"Downloading external file: {file.url}")
-            res = requests.get(file.url, headers = {
-
-                # Set user agent explicitly, so Google Fonts gives us *.woff2
-                # files, which according to caniuse.com is the only format we
-                # need to download as it covers the entire range of browsers
-                # we're officially supporting.
-                "User-Agent": " ".join([
-                    "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
-                    "AppleWebKit/537.36 (KHTML, like Gecko)",
-                    "Chrome/98.0.4758.102 Safari/537.36"
-                ])
-            })
+            try:
+                res = requests.get(
+                    file.url,
+                    headers={
+                        # Set user agent explicitly, so Google Fonts gives us *.woff2
+                        # files, which according to caniuse.com is the only format we
+                        # need to download as it covers the entire range of browsers
+                        # we're officially supporting.
+                        "User-Agent": " ".join(
+                            [
+                                "Mozilla/5.0 (Windows NT 10.0; Win64; x64)",
+                                "AppleWebKit/537.36 (KHTML, like Gecko)",
+                                "Chrome/98.0.4758.102 Safari/537.36",
+                            ]
+                        )
+                    },
+                    timeout=DEFAULT_TIMEOUT_IN_SECS,
+                )
+                res.raise_for_status()
+            except Exception as error:  # this could be a ConnectionError or an HTTPError
+                log.error(f"Could not retrieve {file.url}: {error}")
```

Comment:
Convert this to a warning please. It's idiomatic MkDocs (if there is such a thing) to print this as warnings, so failing the build on those can be opted into with `mkdocs build --strict`.

Reply:
OK, done